smart-web-server/src/main/java/org/smartdata/server/controller/ActionController.java

```diff
@@ -0,0 +1,121 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartdata.server.controller;
+
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import org.smartdata.action.ActionRegistry;
+import org.smartdata.metastore.MetaStoreException;
+import org.smartdata.model.ActionDescriptor;
+import org.smartdata.model.ActionInfo;
+import org.smartdata.model.DetailedFileAction;
+import org.smartdata.server.SmartEngine;
+import org.smartdata.server.engine.CmdletManager;
+import org.springframework.http.MediaType;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import javax.validation.constraints.Min;
+
+import java.io.IOException;
+import java.util.List;
+
+@Tag(name = "Actions")
+@RestController
+@RequestMapping(
+    value = "/api/v1/actions",
+    produces = MediaType.APPLICATION_JSON_VALUE
+)
+@Validated
+public class ActionController {
+  private final CmdletManager cmdletManager;
+
+  public ActionController(SmartEngine smartEngine) {
+    this.cmdletManager = smartEngine.getCmdletManager();
+  }
+
+  @Operation(summary = "List supported actions")
+  @GetMapping("/registry/list")
+  public List<ActionDescriptor> getSupportedActions() throws IOException {
+    return ActionRegistry.supportedActions();
+  }
+
+  @Operation(summary = "List actions")
+  @GetMapping("/list/{pageIndex}/{numPerPage}/{orderBy}/{isDesc}")
```

Comment:
Am I correct that we are currently implementing API contracts with full backwards compatibility? I ask because the current contract seems odd.

Reply:
Yes, currently new server implements old zeppelin REST API